### DISCUSSION TITLE
Adding Pi cam and it is not working

### DISCUSSION THREAD

**Discussion by tekjock** [paukstelis/octoprint_deploy#160] [Q&A]
I setup a Pi Cam on my first RaspberryPi 4. useing Octoprint deploy my new Pi4. I have 2 printers on this Pi and had 2 web cams. one of the cam's died, so, I got a Pi Cam, the same one I have on my other Pi and the feed is not working.

is there some special steps needed to get a Pi cam with Octoprint deploy?

This is the Pi cam I got. - https://www.amazon.com/dp/B06XYDCN5N

**Comment by paukstelis** [comment]
The latest version of octoprint_deploy has an option for adding a PiCam. Two things, however. You will want to switch the streamer to using camera-streamer before you do that. This method has not been tested by me (as I don't have a PiCam to test with).

**Reply by hotdogmang97** [reply]
Hello! first, Thank you for all your work on this. Second, I'm having the same issue with the camera feed streaming to octoprint. I'm using a raspberry pi5 with the picam module 3. Ive tested the camera on the pi 5 using libcam-hello and it is working. I've searched the web for solutions to the "no signal" for the webcam feed on octoprint and haven't found a solution yet. I've read online that the camera-streamer works, but for some reason when I switch from ustreamer to camera-streamer, then the camera-streamer install fails. I'm not sure what to do, because I am completely new to coding, linux, and octoprint. After about a month of trouble shooting and problem solving i've still got no solution. I'm pretty sure my streaming URL is correct and that it would work if I could just get camera-streamer to install on the pi5. Is there some sort of software that i need to install in order to make camera-streamer work on the pi5?
